### DISCUSSION TITLE
including android accessibility sweet in genymotion

### DISCUSSION THREAD

**Discussion by coconut-catalina** [Genymobile/feedback#8] [Genymotion Desktop] [answered]
this is a suggestion, but for the visually impaired who may want to use android on their pc, I would love to see an option to install talkback, you could have a button that says, install android accessibility sweet, recommended for visually impaired users and have it startautomaticly after install, the button could be posissioned the one after installing opengaps, with talkback though, it wouldbe so mcuh easier to sign into the play store, speeking of which, at the moment, google play services does not seem to be working on android 9 and above, but this is primarily about talkback

**Comment by agavignet** [comment, marked as answer]
Thanks for you suggestion, the product team will evaluate it and add it to the roadmap according to other priorities.

Best regards

**Comment by coconut-catalina** [comment]
also, develop an addon for the NVDA screen reader that can let people like
me use genymotion without using complex features. I tried the beta the
other day and feel that you might want to use virtual box imbedded like
other emulators are doing. virtual box on its own just doesn't work
properly with android because the sound is pitched incorrectly. this also
fixes one common issue, unable to start the virtual device. talkback is
open source too so it can easily be included with gaps or in android
altogether as mentioned previously. the current stable version that works
with genymotion is talkback version 8, which is also the maximum supported
by android 5.1

**Reply by agavignet** [reply]
> I tried the beta the
other day and feel that you might want to use virtual box imbedded like
other emulators are doing.

In the beta, we introduced QEMU hypervisor which is still experimental but here to counter various VirtualBox issues. You can give a try :) Settings window > Hypervisor tab > Select QEMU

**Comment by coconut-catalina** [comment]
but as far as I know. the qemu integration only works on linux, and I use
windows as my daily driver. now I love using qemu, I used it with
virt-manager on ubuntu and had no android issues. and I already know qemu
is available for windows but its a pain to set up

**Reply by pgivel** [reply]
In the beta, we ship QEMU with Genymotion. Setting it up is simply a case of going to the genymotion settings and enabling it !

**Comment by coconut-catalina** [comment]
I am pretty sure that the qemu integration only works with mac and linux,
could be wrong though

**Reply by jmaneyrol69** [reply]
QEMU integration works with Windows, but contrary to Linux or Mac, there are indeed some pre-requisite:
1. You need Hyper-V, which may not be enable by default on Windows Pro/Entreprise
2. If you use Windows Home, Hyper-V is not available but it is still possible to use Intel HAXM - however, this will only work with an Intel CPU.

Apart from that, there is nothing to install.

There are still a few bugs too, especially if an app use Webview to connect to the Internet - this will result in major slowdowns. However, we are already working on a fix and the results are encouraging.

In any case, our aim here is to drop VirtualBox altogether in the future; so as soon as QEMU support is better on Windows, it is in your best interest to switch to it.

Regards,

**Comment by coconut-catalina** [comment]
I've had no issues with genymotion and qemu under linux, orca somewhat
works when navigating, as for windows, I get no sound on qemu what so ever,
about talkback, since you still offer android lollipop, its probably worth
offering a version of talkback that works with all the versions of android
you offer, ie, talkback 7, considering thats the max supported by android
5.1, the latest version of talkback requires android nougat and up

**Reply by jmaneyrol69** [reply]
>I get no sound on qemu what so ever

Hence "experimental" :slightly_smiling_face: 

However, it would be interesting for us to investigate. Would you mind sharing Genymotion logs with us?
You can refer to the following article if you don't know how to generate them: https://support.genymotion.com/hc/en-us/articles/360002784898-How-to-get-Genymotion-Desktop-logs-

Thanks

**Comment by coconut-catalina** [comment]
I'm now running linux on that computer, and unlike windows, genymotion is
in fully working order, including android 10, now i wonder how I turn my
talkback and android setup wizard launch batch files into linux shell
scripts, somehow, I don't think renaming them and changing the extension to
a .sh will work as I made them with windows in mind

**Reply by jmaneyrol69** [reply]
Ok. Would you mind sharing your batch files?

**Comment by coconut-catalina** [comment]
will have to share them separately, first off, the talkback one, let me
know if you want the android setup wizard file.
https://1drv.ms/u/s!AiPxrC3Ad0KjipMbPksIYsRVLrfhDA?e=uNvGH5

**Reply by jmaneyrol69** [reply]
That one's easy to convert into bash shell (.sh). You can try this code:

```bash
#!/bin/bash

adb shell settings put secure enabled_accessibility_services com.google.android.marvin.talkback/com.google.android.marvin.talkback.TalkBackService
```

If you haven't set the path to adb in your environment PATH variable, or in `.bashrc`, just add the path before "adb".
Oh and don't forget to make your script executable with `chmod +x your_script.sh`

**Comment by coconut-catalina** [comment]
saying that I've had a littel deal of success getting batch files to work
under wine, although my preferred distribution still doesn't provide the
latest stable version of wine, even though we are on ubuntu 22.04 and a
version of wine that has since reached end of life

**Reply by jmaneyrol69** [reply]
If all your batch files simply run adb commands, it should not be too difficult to convert them into shell scripts.

**Reply by jmaneyrol69** [reply]
If you aren't using a separate Android SDK, you can still use Genymotion built-in adb - it should be located in `$HOME/genymotion/tools`
